tutorials/shading/visual_shader.rst

```diff
@@ -0,0 +1,101 @@
+- The ``Expand All`` and ``Collapse All`` options in the drop-down option menu can be used to easily list the available nodes.
+
+When connecting any ``scalar`` output to ``vector`` input, all the 3 components of the vector will be the value of the scalar.
+
+When connecting any ``vector`` output to ``scalar`` input, the value of the scalar will be the dot product between the original vector and a Vector3(0.33,0.33,0.33).
+
+The popup has categorized the nodes, it is overwhelming to understand or use most of them, try to pick some of them, plug it into the output socket and observe what happens.
+
+.. note::
+    
+   Although ``VisualShader`` does not require coding, it shares the same logic with normal shaders, you can learn and convert shaders to visual shaders after understanding the logic behind the shader.
+
+Below are some special nodes that are worth knowing.
+
+Expression node
++++++++++++++++
+
+The ``Expression`` node allows you to write Godot Shading Language (GLSL-like) expressions inside your visual shaders! The node has buttons to add any amount of required input and output ports and can be resized. You can also set up the name and type of each port. The expression you have entered will apply immediately (after focus leaves out the expression text box) to the material, any parsing or compilation errors will be printed to the Output tab. The outputs are initialized to their zero value by default. The new node is located under the Special tab and can be used in all shader modes.
+
+.. image:: img/vs_expression.gif
+
+The possibilities of this node are almost limitless – you can write complex procedures, and use all the power of Godot (text-based) shaders, such as loops, discard keyword, extended types, etc. For example:
+
+.. image:: img/vs_expression2.png
+
+Fresnel node
+++++++++++++
+
+The ``Fresnel`` node is designed to accept normal and view vectors and produces a scalar which is the saturated dot product between them. Additionally you can setup the inversion and the power of equation
+
+.. image:: img/vs_fresnel.png
+
+Boolean node
+++++++++++++
+
+The ``Boolean`` node can be converted to Scalar or Vector to represent 0 or 1 and (0, 0, 0) or (1, 1, 1) respectively. This property can be used to enable or disable some effect parts by one click.
+
+.. image:: img/vs_boolean.gif
+
+If node
++++++++
+
+The ``If`` node allows you to setup a vector which will be returned after performing comparison between a and b. There are three vectors which can be returned: a == b (in that case the tolerance parameter is provided as a comparison threshold – by default it is equal to the minimal value, e.g. 0.00001), a > b and a < b.
+
+.. image:: img/vs_if.png
+
+Switch node
++++++++++++
+
+The ``Switch`` node returns a vector if the boolean condition is true or false. ``Boolean`` was introduced above. If you convert a vector to a true boolean, all components of the vector should be above zero.
+
+.. image:: img/vs_switch.png
+
```

Comment:
add:

.. note::
    The ``switch`` node is only available on the GLES3 backed. If you are targeting GLES2 devices, you cannot use ``switch`` statements.